engine/src/main/java/org/terasology/engine/subsystem/common/TelemetrySubSystem.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.engine.subsystem.common;
+
+import com.snowplowanalytics.snowplow.tracker.emitter.BatchEmitter;
+import com.snowplowanalytics.snowplow.tracker.emitter.Emitter;
+import com.snowplowanalytics.snowplow.tracker.emitter.RequestCallback;
+import com.snowplowanalytics.snowplow.tracker.http.ApacheHttpClientAdapter;
+import com.snowplowanalytics.snowplow.tracker.http.HttpClientAdapter;
+import com.snowplowanalytics.snowplow.tracker.payload.TrackerPayload;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.config.Config;
+import org.terasology.context.Context;
+import org.terasology.engine.subsystem.EngineSubsystem;
+import org.terasology.telemetry.Metrics;
+import org.terasology.telemetry.TelemetryParams;
+import org.terasology.telemetry.logstash.TelemetryLogstashAppender;
+
+import java.util.List;
+
+import static org.terasology.telemetry.TelemetryUtils.fetchTelemetryLogstashAppender;
+
+/**
+ * This is a telemetry engine system.
```

Comment:
Maybe a slightly longer description, eventually with a link or two to documentation somewhere? :-)